pkg/controller/migplan/gvk.go

```diff
@@ -0,0 +1,414 @@
+package migplan
+
+import (
+	"strings"
+
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var crdGVR = schema.GroupVersionResource{
+	Group:    "apiextensions.k8s.io",
+	Version:  "v1beta1", // Should become v1 after 1.17, needs downscaling
+	Resource: "customresourcedefinitions",
+}
+
+// Compare is a store for discovery and dynamic clients to do GVK compare
+type Compare struct {
+	Plan         *migapi.MigPlan
+	SrcDiscovery discovery.DiscoveryInterface
+	DstDiscovery discovery.DiscoveryInterface
+	SrcClient    dynamic.Interface
+}
+
+func (r ReconcileMigPlan) compareGVK(plan *migapi.MigPlan) error {
+	// No spec chage this time
+	if plan.HasReconciled() || !clustersReady(plan) {
+		return nil
+	}
+
+	gvkCompare, err := r.newGVKCompare(plan)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	unsupportedMapping, err := gvkCompare.Compare()
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	reportGVK(plan, unsupportedMapping)
+
+	return nil
+}
+
+func (r ReconcileMigPlan) newGVKCompare(plan *migapi.MigPlan) (*Compare, error) {
+	gvkCompare := &Compare{
+		Plan: plan,
+	}
+
+	err := gvkCompare.NewSourceDiscovery(r)
+	if err != nil {
+		return nil, err
+	}
+
+	err = gvkCompare.NewDestinationDiscovery(r)
+	if err != nil {
+		return nil, err
+	}
+
+	err = gvkCompare.NewSourceClient(r)
+	if err != nil {
+		return nil, err
+	}
+
+	return gvkCompare, nil
+}
+
+func reportGVK(plan *migapi.MigPlan, unsupportedMapping map[string][]schema.GroupVersionResource) {
+	unsupportedNamespaces := []migapi.UnsupportedNamespace{}
```

Comment:
`incompatible` would be a more accurate term across the board in this PR. Please change all instances of `unsupported` to `incompatible`